src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -33,9 +37,15 @@ public abstract class FormulaType
         public static FormulaType OptionSetValue { get; } = new OptionSetValueType();
 
         // chained by derived type 
-        internal FormulaType(DType type)
+        internal FormulaType(DType type, DisplayNameProvider displayNameProvider = null)
         {
-            _type = type;
+            if (displayNameProvider == null)
+                _type = type;
+            else 
+            {
+                _displayNameProvider = displayNameProvider;
+                _type = DType.AttachDisplayNameProvider(type, displayNameProvider);
+            }
```

Comment:
to consider - this display name should already be associated with the incoming Dtype. 

Else we break this invariant:
DType t;
new FormulaType(t)._type == t.

Reply:
ie, keep FormulaType as a lightweight wrapper over Dtype and avoid logic leaking into it.

Reply:
Great point, fixed